iroh-store/src/store.rs

```diff
@@ -366,11 +367,14 @@ impl<'a> WriteStore<'a> {
         let mut total_blob_size = 0;
 
         let mut batch = WriteBatch::default();
+        let mut cid_tracker: HashSet<Cid> = std::collections::HashSet::default();
         for (cid, blob, links) in blocks.into_iter() {
-            if self.has(&cid)? {
-                return Ok(());
+            if self.has(&cid)? || cid_tracker.contains(&cid) {
```

Comment:
Not that it matters that much, but do the cid_tracker membership test first. It is much faster than having to go all the way to the DB.

Reply:
love these little pieces of advice!